KoPrivateGPT/retrieval/base.py

```diff
@@ -18,3 +25,84 @@ def ingest(self, passages: List[Passage]):
     @abstractmethod
     def retrieve_id(self, query: str, top_k: int = 5, *args, **kwargs) -> List[Union[str, UUID]]:
         pass
+
+    # If you want more db type, you can add more db type below this line.
+    @staticmethod
+    def split_db_type(ids: List[UUID], db_origin_list):
+        mongo_db_ids = []
+        mongo_db_path = []
+        pickle_db_ids = []
+        pickle_db_path = []
+        for i in range(len(db_origin_list)):
+            if db_origin_list[i]['db_type'] == "mongo_db":
+                mongo_db_ids.append(ids[i])
+                mongo_db_path.append(db_origin_list[i]['db_path'])
+            elif db_origin_list[i]['db_type'] == "pickle_db":
+                pickle_db_ids.append(ids[i])
+                pickle_db_path.append(db_origin_list[i]['db_path'])
+            else:
+                raise ValueError(f"Unknown db type: {db_origin_list[i]['db_type']}")
+        return mongo_db_ids, mongo_db_path, pickle_db_ids, pickle_db_path
+
+    def fetch_data(self, ids: List[UUID]) -> List[Passage]:
+        self.redis_db = RedisDBSingleton()
+        db_origin_list = self.redis_db.get_json(ids)
+        # Sometimes redis doesn't find the id, so we need to filter that db_origin is None.
```

Comment:
You must make proper error try-catch code here. The reason that redis can't find the id might be redis DB's error, or data not properly registered at redis linker. It can be critical error. I think there is a better way to catch this kind of error.